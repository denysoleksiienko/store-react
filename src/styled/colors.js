export const COLORS = {
  white: '#fff',
  orchid: '#9932cc',
  solitude: '#e9ecef',
  gray: '#808080',
};
