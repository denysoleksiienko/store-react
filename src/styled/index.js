export { RowWrap, ColOrders } from './App';
export { BreadcrumbWrap, NavLinkStyled } from './Breadcrumbs';
export { StyledButton } from './Button';
export { SecureInfo, Icon, Span } from './Payment';
export { InputSpan } from './ContactFields';
export { OrderWrap } from './FinishOrders';
export {
  SummaryTitle,
  SummaryWrap,
  SummaryList,
  SummaryItem,
  ProductInfo,
  SummaryCost,
  SummaryTotal,
  SummaryCostInner,
} from './Orders';

export { Title } from './shared/Title';
export { Label } from './shared/Label';
export { InputsWrap } from './shared/InputsWrap';
export { FormGroupInner } from './shared/FormGroupInner';
