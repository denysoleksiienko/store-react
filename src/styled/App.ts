import styled from 'styled-components';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { COLORS } from '../constants/colors';

export const RowWrap = styled(Row)`
  height: 70vh;
`;

export const ColWrap = styled(Col)`
  background: ${COLORS.solitude};
`;
