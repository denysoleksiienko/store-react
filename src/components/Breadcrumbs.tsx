import React from 'react';
import Breadcrumb from 'react-bootstrap/Breadcrumb';
import { BreadcrumbWrap, NavLinkStyled } from 'styled/Breadcrumbs';

export const Breadcrumbs: React.FC = () => (
  <BreadcrumbWrap>
    <Breadcrumb.Item linkAs={NavLinkStyled} linkProps={{ to: '/', exact: true }}>
      Shipping
    </Breadcrumb.Item>
    <Breadcrumb.Item linkAs={NavLinkStyled} linkProps={{ to: '/billing' }}>
      Billing
    </Breadcrumb.Item>
    <Breadcrumb.Item linkAs={NavLinkStyled} linkProps={{ to: '/payment' }}>
      Payment
    </Breadcrumb.Item>
  </BreadcrumbWrap>
);
