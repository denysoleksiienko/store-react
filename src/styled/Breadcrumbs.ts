import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import Breadcrumb from 'react-bootstrap/Breadcrumb';

import { COLORS } from '../constants/colors';

export const BreadcrumbWrap = styled(Breadcrumb)`
  ol.breadcrumb {
    background: ${COLORS.white};
  }
  color: ${COLORS.orchid};
`;

export const NavLinkStyled = styled(NavLink)`
  font-weight: bold;
  color: ${COLORS.gray};
  &.active {
    color: ${COLORS.orchid};
  }
`;
