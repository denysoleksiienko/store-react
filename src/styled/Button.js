import styled from 'styled-components';
import Button from 'react-bootstrap/Button';
import { COLORS } from '../constants/colors';

export const StyledButton = styled(Button)`
  background-color: ${COLORS.orchid};
  width: 100%;
`;
