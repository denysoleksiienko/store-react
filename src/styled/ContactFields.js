import styled from 'styled-components';

export const InputSpan = styled.span`
  margin-left: 20px;
  font-size: 12px;
`;
