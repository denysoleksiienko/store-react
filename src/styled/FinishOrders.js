import styled from 'styled-components';

export const OrderWrap = styled.div`
  margin-top: 50px;
  margin-left: 16px;
`;
