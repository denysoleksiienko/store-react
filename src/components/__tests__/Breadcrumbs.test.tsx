import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, ShallowWrapper } from 'enzyme';

import Breadcrumb from 'react-bootstrap/Breadcrumb';
import { Breadcrumbs } from '../Breadcrumbs';
import { BreadcrumbWrap } from 'styled/Breadcrumbs';

const setUp = () => shallow(<Breadcrumbs />);

describe('Breadcrumbs component', () => {
  let component: any[] | ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    component = setUp();
  });

  it('should be defined', () => {
    const wrapper = shallow(<Breadcrumbs />);
    expect(wrapper).toBeDefined();
  });

  it('should contain wrapper with styled component', () => {
    const wrapper = component.find(BreadcrumbWrap);
    expect(wrapper.length).toBe(1);
  });

  it('should contain three links', () => {
    const links = component.find(Breadcrumb.Item);

    expect(links.length).toBe(3);
  });

  it('should match snapshot', () => {
    const wrapper = renderer
      .create(
        <Router>
          <Breadcrumbs />
        </Router>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
