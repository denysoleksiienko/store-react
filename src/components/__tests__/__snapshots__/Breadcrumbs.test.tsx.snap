// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should match snapshot 1`] = `
<nav
  aria-label="breadcrumb"
  className="sc-bdfBwQ jXuzUi"
>
  <ol
    className="breadcrumb"
  >
    <li
      className="breadcrumb-item"
    >
      <a
        aria-current="page"
        className="sc-gsTCUz ffWKEV active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Shipping
      </a>
    </li>
    <li
      className="breadcrumb-item"
    >
      <a
        aria-current={null}
        className="sc-gsTCUz ffWKEV"
        href="/billing"
        onClick={[Function]}
      >
        Billing
      </a>
    </li>
    <li
      className="breadcrumb-item"
    >
      <a
        aria-current={null}
        className="sc-gsTCUz ffWKEV"
        href="/payment"
        onClick={[Function]}
      >
        Payment
      </a>
    </li>
  </ol>
</nav>
`;
